File: apzkr-pzpi-21-4-bondarenko-taras/Task4-MobileClient/CargoTruckMobileClient/app/src/main/java/com/example/cargotruckmobile/Models/Cargo.kt
package com.example.cargotruckmobile.Models


data class Cargo(
    val id: String,
    val name: String,
    val weight: Float,
    val length: Float,
    val height: Float,
    val width: Float,
    val announcedPrice: Float,
    val shippingPrice: Float,
    val noticeId: String?,
    val senderId: String,
    val receiverId: String,
    val citySenderId: String,
    val addressSenderId: String,
    val cityReceiverId: String,
    val addressReceiverId: String,
    val status: String
)
data class CargoListResponse(
    val items: List<Cargo>,
    val pageNumber: Int,
    val pageSize: Int,
    val totalPages: Int,
    val totalItems: Int,
    val hasPreviousPage: Boolean,
    val hasNextPage: Boolean
)
data class CargoCreateDto(
    val name: String,
    val weight: Float,
    val length: Float,
    val height: Float,
    val width: Float,
    val announcedPrice: Float,
    val senderId: String,
    val receiverId: String,
    val citySenderId: String,
    val addressSenderId: String,
    val cityReceiverId: String,
    val addressReceiverId: String
)
data class DetailsCargo(
    val id: String,
    val name: String,
    val weight: Int,
    val length: Int,
    val height: Int,
    val width: Int,
    val announcedPrice: Int,
    val shippingPrice: Int,
    val noticeId: String?,
    val senderId: String,
    val receiverId: String,
    val citySenderId: String,
    val addressSenderId: String,
    val cityReceiverId: String,
    val addressReceiverId: String,
    val status: String
)

data class CargoUpdateNotice(
    val cargoId: String,
    val newNoticeId: String,

    )

File: apzkr-pzpi-21-4-bondarenko-taras/Task3-WebClient/CargoTruckWebClient/src/Models/Cargo.js
export class Cargo {
    constructor(
      id,
      name,
      weight,
      length,
      height,
      width,
      announcedPrice,
      shippingPrice,
      noticeId,
      senderId,
      receiverId,
      citySenderId,
      addressSenderId,
      cityReceiverId,
      addressReceiverId,
      status
    ) {
      this.id = id;
      this.name = name;
      this.weight = weight;
      this.length = length;
      this.height = height;
      this.width = width;
      this.announcedPrice = announcedPrice;
      this.shippingPrice = shippingPrice;
      this.noticeId = noticeId;
      this.senderId = senderId;
      this.receiverId = receiverId;
      this.citySenderId = citySenderId;
      this.addressSenderId = addressSenderId;
      this.cityReceiverId = cityReceiverId;
      this.addressReceiverId = addressReceiverId;
      this.status = status;
    }
  }
  
  export class CargoCreateDto {
    constructor(
      name,
      weight,
      length,
      height,
      width,
      announcedPrice,
      senderId,
      receiverId,
      citySenderId,
      addressSenderId,
      cityReceiverId,
      addressReceiverId
    ) {
      this.name = name;
      this.weight = weight;
      this.length = length;
      this.height = height;
      this.width = width;
      this.announcedPrice = announcedPrice;
      this.senderId = senderId;
      this.receiverId = receiverId;
      this.citySenderId = citySenderId;
      this.addressSenderId = addressSenderId;
      this.cityReceiverId = cityReceiverId;
      this.addressReceiverId = addressReceiverId;
    }
  }
  